js/modals/question-modal.js

```diff
@@ -85,96 +85,104 @@ $(function() {
-      var io3 = '<button type="button" class="btn btn-default io" data-io="3"><span>3</span></button>';
-      var $inputTextarea = $('<textarea class="form-control" id="input-textarea" rows="3" placeholder="Question Input 1"></textarea>');
-      var $outputInput = $('<input type="text" id="output-input" class="form-control" placeholder="Expected Output 1">');
-
-      $io
-          .append($removeIo)
-          .append($addIo)
-          .append(io1)
-          .append(io2)
-          .append(io3)
-          .append($inputTextarea)
-          .append($outputInput);
+
+      $('.js-question-type-area').attr('class', 'js-question-type-area coding');
+
+      var $io = $templates.find('.js-modal-question-code .js-io')
+          .clone()
+          .children();
 
       return $editor.add($io);
    }
 
-   function ioButtonClick(button) {
-      if (!button.hasClass('active')) {
-         $('fieldset.input-output .io.active').data('input', $('fieldset.input-output textarea').val());
-         $('fieldset.input-output .io.active').data('output', $('fieldset.input-output input').val());
-         $('fieldset.input-output textarea').val(button.data('input'));
-         $('fieldset.input-output input').val(button.data('output'));
-         $('fieldset.input-output textarea').attr('placeholder', 'Question Input ' + button.data('io'));
-         $('fieldset.input-output input').attr('placeholder', 'Expected Output ' + button.data('io'));
-         $('.io.active>span').text($('.io.active').data('io'));
-         $('.io.active').removeClass('active');
-
-         button
-             .addClass('active')
+   function ioButtonClick($button) {
+      if (!$button.hasClass('active')) {
+         var $activeButton = $('.modal .js-io-active');
+         var $ioTextarea = $('.modal .js-io-input');
+         var $ioOutput = $('.modal .js-io-output');
+
+         $activeButton
+             .data('input', $ioTextarea.val())
+             .data('output', $ioOutput.val())
+             .removeClass('js-io-active')
+             .find('span')
+             .text($activeButton.data('io'));
+
+         $ioTextarea
+             .val($button.data('input'))
+             .attr('placeholder', 'Question Input ' + $button.data('io'));
+         $ioOutput
+             .val($button.data('output'))
+             .attr('placeholder', 'Expected Output ' + $button.data('io'));
+
+         $button
+             .addClass('js-io-active')
              .find('span')
-             .text('IO ' + button.data('io'));
+             .text('IO ' + $button.data('io'));
 
-         $('fieldset.input-output textarea').focus();
+         $ioTextarea.focus();
       }
    }
 
    function bindCodeAreaEvents() {
-      $('.modal').on('click', '.remove-io', function() {
-         if ($('.io').length > 3) {
-            $('.io').last().remove();
-         }
-      });
+      var $qa = $('#q-and-a-plugin');
```

Comment:
To improve readability it's better to avoid abbreviations. It's better to have  something like:

`var $questionInput = $('#elem');`

than

`var $qInput = $('#elem');`